system/core/tags.php

```diff
@@ -0,0 +1,184 @@
+<?php
+	// TODO:
+	// - add styling for default theme
+	// - implement error notification if new tag exists
+	// - implement tag management page: edit, translate, remove or merge tags
+	// - implement tagcontainer pagetype
+	
+	function tagList($pageListNode, $lang) {
+		global $hyphaXml;
+		
+		// Build list of active tags for given page
+		foreach($pageListNode->getElementsByTagName('tag') as $tag) $activeTags[] = $tag->getAttribute('id');
```

Comment:
I believe getElementsByTagName returns all descendent tags, not just direct children. Is that intended here? It probably doesn't hurt, though.

Reply:
Kinda intended. Using children would add the necessity to check if the childnode is has 'tag' as tagname. This could be done but adds little in terms of performance or robustness. Lacking a getChildrenByTagName function I chose this approach.

Reply:
This could be easily done using `find` and an xpath expression, I think.

Reply:
Laurens fixed this with XPath.